### PR TITLE
Update compatibility matrix

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Update compatibility matrix as per the actual compatibility.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,10 +85,11 @@ Refer to [CONTRIBUTING](./CONTRIBUTING.md) for build instructions.
...
 Nessie Iceberg's integration is compatible with Iceberg as in the following table:
 
-| Nessie version | Iceberg version | Spark version | Hive version | Flink version |
-|----------------|-----------------|---------------|--------------|---------------|
-| 0.22.0         | 0.13.1          | 3.0.X, 3.1.X  | 2.3.9        | 1.12.1        |
-| 0.9.2          | 0.12.1, 0.12.0  | 3.0.X, 3.1.X  | 2.3.9        | 1.12.1        |
+| Nessie version | Iceberg version | Spark version      | Hive version | Flink version |
+|----------------|-----------------|--------------------|--------------|---------------|
+| 0.22.0         | upcoming release| 3.0.X, 3.1.X, 3.2.X| 2.3.9        | 1.12.1        |
+| 0.19.0         | 0.13.1          | 3.0.X, 3.1.X       | 2.3.9        | 1.12.1        |
```
Comment (ajantha-bhat) [inline README.md:91]: In `0.20.0` we broke compatibility in https://github.com/projectnessie/nessie/pull/3226 (only in negative scenarios, for error handling)

and we again broke it in `0.21.1` with this change https://github.com/projectnessie/nessie/pull/3431

**Review by nastra — APPROVED** [review]

**Review by snazy — CHANGES_REQUESTED** [review]
Waiting for clarification how the mentioned changes broke Iceberg 0.13.1 running against Nessie server 0.22.0/0.21.1.

**Comment by ajantha-bhat** [comment]
>Waiting for clarification how the mentioned changes broke Iceberg 0.13.1 running against Nessie server 0.22.0/0.21.1.

"Show log" from sql extension throws error about arguments mismatch for
StreamingUtil.getCommitLogStream
as it is used in BaseShowLogExec.scala

**Comment by ajantha-bhat** [comment]
Even Use reference command is broken
image

**Comment by snazy** [comment]
> Even Use reference command is broken image

You're mixing Nessie 0.22.0 classes into icberg-runtime uber jar maybe?

**Comment by ajantha-bhat** [comment]
```
/Users/ajantha/Documents/tools/spark-3.0.3-bin-hadoop2.7/bin/spark-sql --packages org.slf4j:slf4j-api:1.7.32 --jars /Users/ajantha/Downloads/iceberg-spark3-runtime-0.13.1.jar,/Users/ajantha/Downloads/nessie-spark-extensions-0.22.0.jar \
                                    --conf spark.sql.catalog.nessie=org.apache.iceberg.spark.SparkCatalog \
                                    --conf spark.sql.catalog.nessie.warehouse=$PWD/nessie2 \
                                --conf spark.sql.catalog.nessie.catalog-impl=org.apache.iceberg.nessie.NessieCatalog \
                                --conf spark.sql.catalog.nessie.uri=http://localhost:19120/api/v1 \
                                --conf spark.sql.catalog.nessie.ref=main \
                                --conf spark.sql.catalog.nessie.auth-type=NONE \
                               --conf spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions
```

I am using the latest Nessie (0.22.0) spark extensions with released Iceberg 0.13.1.
What version of Nessie spark extensions I should use? Isn't the compatibility also includes sql extension module ?

**Comment by snazy** [comment]
> I am using the latest Nessie (0.22.0) spark extensions with released Iceberg 0.13.1.

This cannot work, right? You are mixing Nessie classes from 0.18.0 included in the iceberg-runtime uber-jar with *conflicting* classes in Nessie 0.22.0. Either you use iceberg-runtime as it is or plain Nessie.

**Comment by ajantha-bhat** [comment]
> This cannot work, right? You are mixing Nessie classes from 0.18.0 included in the iceberg-runtime uber-jar with conflicting classes in Nessie 0.22.0. Either you use iceberg-runtime as it is or plain Nessie.

Nessie sql extension module depends on `client.nessie.version` which is still 0.18.0, so it should work and it is working now.

It seems my local environment had Iceberg snapshot jar instead of iceberg jar. Hence it was throwing this error. I will close this. apologies for the noise.

**Comment by ajantha-bhat** [comment]
But with 0.13.1 Iceberg run time jar, we cannot use 0.18.0 Nessie sql extension as it is pointing to `client.nessie.version` 0.9.2 which brings conflicting classes. I don't know where to mention this compatibility.

**Pull request closed without merging**